pkg/api/defaults-custom-cloud-profile.go

```
@@ -3,34 +3,113 @@
 
 package api
 
-import "github.com/Azure/aks-engine/pkg/helpers"
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strings"
+	"time"
 
-func (p *Properties) setCustomCloudProfileDefaults() {
+	"github.com/Azure/aks-engine/pkg/helpers"
+	"github.com/Azure/go-autorest/autorest/azure"
+)
 
+func (cs *ContainerService) setCustomCloudProfileDefaults() {
+	p := cs.Properties
 	if p.IsAzureStackCloud() {
+		p.CustomCloudProfile.AuthenticationMethod = helpers.EnsureString(p.CustomCloudProfile.AuthenticationMethod, ClientSecretAuthMethod)
+		p.CustomCloudProfile.IdentitySystem = helpers.EnsureString(p.CustomCloudProfile.IdentitySystem, AzureADIdentitySystem)
+		p.CustomCloudProfile.DependenciesLocation = DependenciesLocation(helpers.EnsureString(string(p.CustomCloudProfile.DependenciesLocation), AzureStackDependenciesLocationPublic))
+		p.SetAzureStackCloudSpec()
+		err := cs.SetCustomCloudProfileEnvironment()
+		if err != nil {
+			log.Printf("Failed to set environment - %s", err)
```

this should return an error instead of logging it